src/dash-table/conditional/index.ts

```diff
@@ -82,4 +85,36 @@ export function ifEditable(condition: IEditableElement | undefined, isEditable:
             return true;
     }
     return isEditable === condition.column_editable;
-}
\ No newline at end of file
+}
+
+export type Filter<T> = (s: T[]) => T[];
+
+export const matchesDataCell = (datum: Datum, i: number, column: IVisibleColumn): Filter<IConvertedStyle> => R.filter<IConvertedStyle>((style =>
+    style.matchesRow(i) &&
+    style.matchesColumn(column) &&
+    style.matchesFilter(datum)
+));
+
+export const matchesFilterCell = (column: IVisibleColumn): Filter<IConvertedStyle> => R.filter<IConvertedStyle>((style =>
+    style.matchesColumn(column)
+));
+
+export const matchesHeaderCell = (i: number, column: IVisibleColumn): Filter<IConvertedStyle> => R.filter<IConvertedStyle>((style =>
+    style.matchesRow(i) &&
+    style.matchesColumn(column)
+));
+
+export const matchesDataOpCell = (datum: Datum, i: number): Filter<IConvertedStyle> => R.filter<IConvertedStyle>((style =>
+    !style.checksColumn() &&
+    style.matchesRow(i) &&
+    style.matchesFilter(datum)
+));
+
+export const getFilterOpStyles: Filter<IConvertedStyle> = R.filter<IConvertedStyle>((style =>
+    !style.checksColumn()
+));
+
+export const getHeaderOpStyles = (i: number): Filter<IConvertedStyle> => R.filter<IConvertedStyle>((style =>
+    style.matchesRow(i) &&
+    !style.checksColumn()
+));
```

Comment:
Flavored filter helpers for various usage scenarios (data, dataOp, filter, etc.) -- used as building blocks by edges and style calcs